### ISSUE TITLE
custom output destination

### ISSUE THREAD

**Issue by nikhedonia** [strace/strace#82]
One of my projects uses strace extensively generating logs of up to 100GB In order to perform quite time intensive analysis. We would benefit tremendously if we could make a modified version of strace that performs some preprocessing and write to an database.

Ideally we would use strace as an [library](https://github.com/strace/strace/issues/70) but I don't understand strace well enough to perform this modifications yet.

So I'm wondering, could someone point me in the relevant source locations to patch strace so it can write to eg. a custom destination(eg. a database)?

If you think there is any value to add an option to strace to load a user-supplied callback from a shared-library I'd be happy to make a PR once I manage with your help to make the appropriate modifications

**Comment by esyr** [comment]
You can try to leverage -o| option (piping output to a program).

**Comment by nikhedonia** [comment]
@esyr this seems clunky if used in conjunction with "-f".

```
4226  17:26:49 <... openat resumed> )   = -1 ENOENT (No such file or directory)
4230  17:26:49 <... close resumed> )    = 0
```

is there any way to tell strace to combine `suspended...` and `...resume` statements, or do I have to parse and recombine it myself?

**Comment by esyr** [comment]
As of now, it's not possible to use -ff and output piping simultaneously.

**Comment by esyr** [comment]
Patch in #49 implements stashing of syscall entering output, and that likely can be used for output without continuations.

**Comment by ldv-alt** [comment]
Yes, the side effect of --status option can be used to avoid unfinished/resumed lines in the log.
